app/controllers/api/v2/application_controller.rb

```diff
@@ -54,7 +55,11 @@ def append_info_to_payload(payload)
             sign_in_id: current_user.sign_in_user_id,
           }
         end
-        payload[:request_id] = request.uuid
+        payload[:request_id] = RequestStore.store[:request_id]
+      end
+
+      def store_request_id
+        RequestStore.store[:request_id] = request.uuid || SecureRandom.uuid
```

Comment:
Do we need the `|| SecureRandom.uuid`? So far as I can tell, if there is no request-id one will be generated for us.

Reply:
I added this based on the acceptance criteria but looks like you are correct judging from this => https://api.rubyonrails.org/classes/ActionDispatch/RequestId.html

Reply:
Have removed :-)